dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,795 @@
+        }
+
+        return Task.WhenAll(tasks);
+    }
+
+    /// <inheritdoc />
+    public Task<VectorSearchResults<TRecord>> VectorizedSearchAsync<TVector>(TVector vector, VectorSearchOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        const string LimitPropertyName = "k";
+
+        Verify.NotNull(vector);
+
+        var vectorType = vector.GetType();
+
+        if (!SqliteConstants.SupportedVectorTypes.Contains(vectorType))
+        {
+            throw new NotSupportedException(
+                $"The provided vector type {vectorType.FullName} is not supported by the SQLite connector. " +
+                $"Supported types are: {string.Join(", ", SqliteConstants.SupportedVectorTypes.Select(l => l.FullName))}");
+        }
+
+        var searchOptions = options ?? s_defaultVectorSearchOptions;
+        var vectorProperty = this.GetVectorPropertyForSearch(searchOptions.VectorPropertyName);
+
+        if (vectorProperty is null)
+        {
+            throw new InvalidOperationException("The collection does not have any vector properties, so vector search is not possible.");
+        }
+
+        var mappedArray = SqliteVectorStoreRecordPropertyMapping.MapVector(vector);
+
+        // Simulating skip/offset logic locally, since OFFSET can work only with LIMIT in combination
+        // and LIMIT is not supported in vector search extension, instead of LIMIT - "k" parameter is used.
+        var limit = searchOptions.Top + searchOptions.Skip;
+
+        var conditions = new List<SqliteWhereCondition>()
+        {
+            new SqliteWhereMatchCondition(this._propertyReader.StoragePropertyNamesMap[vectorProperty.DataModelPropertyName], mappedArray),
+            new SqliteWhereEqualsCondition(LimitPropertyName, limit)
+        };
+
+        var filterConditions = this.GetFilterConditions(searchOptions.Filter, this._dataTableName);
+
+        if (filterConditions is { Count: > 0 })
+        {
+            conditions.AddRange(filterConditions);
+        }
+
+        var vectorSearchResults = new VectorSearchResults<TRecord>(this.EnumerateAndMapSearchResultsAsync(
+            conditions,
+            searchOptions,
+            cancellationToken));
+
+        return Task.FromResult(vectorSearchResults);
+    }
+
+    #region Implementation of IVectorStoreRecordCollection<ulong, TRecord>
+
+    /// <inheritdoc />
+    public Task<TRecord?> GetAsync(ulong key, GetRecordOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        return this.InternalGetAsync(key, options, cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<TRecord> GetBatchAsync(IEnumerable<ulong> keys, GetRecordOptions? options = null, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        await foreach (var record in this.InternalGetBatchAsync(keys, options, cancellationToken).ConfigureAwait(false))
```

Comment:
We should be able to just return the result of `this.InternalGetBatchAsync` without the loop here, since the return types are the same.